uint64[] word_ids
float64[] word_values